src/envoy/http/authn/authn_utils.cc

```diff
@@ -22,47 +22,31 @@ namespace Http {
 namespace Istio {
 namespace AuthN {
 namespace {
-class JsonIterCallback : public Logger::Loggable<Logger::Id::filter> {
- public:
-  // The json iterator callback function
-  static bool JsonIteratorCallback(const std::string&,
-                                   const Envoy::Json::Object&,
-                                   istio::authn::JwtPayload* payload);
-};
-
-bool JsonIterCallback::JsonIteratorCallback(const std::string& key,
-                                            const Envoy::Json::Object& obj,
-                                            istio::authn::JwtPayload* payload) {
-  ENVOY_LOG(debug, "{}: key is {}", __FUNCTION__, key);
-  ::google::protobuf::Map< ::std::string, ::std::string>* claims =
-      payload->mutable_claims();
+// Extract JWT audience into the JwtPayload.
+// Return true if "aud" is extracted. Otherwise, return false.
+bool ExtractJwtAudience(const Envoy::Json::Object& obj,
+                        istio::authn::JwtPayload* payload) {
+  const std::string key = "aud";
+  // "aud" can be either string array or string.
+  // First, try as string, will throw execption if object type is not string.
+  try {
```

Comment:
You can check claims["aud"] instead, as if it a string, it should be there already. It saves a try catch, which typically more expensive.

Reply:
Using checking claims["aud"] instead assumes that ExtractJwtAudience() is called after extracting claims. Such calling order requirement tends to be error prone.